### PR TITLE
Review of my Twitter Clone

### PR DESCRIPTION
**Pull request by luqmanmusah**
I have not yet completed all the required features but I would like to get some review on my capstone as I continue to work on it.

I was supposed to meet all the requirements mention below but I was able to meet about 70% but I have spent over 10 days due to some personal problems.

I have learned a great deal working on this capstone.

- [x]  The user logs in to the app, only by typing the username (a proper authenticated login is **not** a requirement).

- [x]  The user is presented with the homepage (see the *Homepage* screenshot above) that includes:

    1. Left-side menu (includes only links to pages that are implemented).
    2. *Tweets* tab in the centre (skip *Photos* and *Videos* for this MVP).
    3. Right-side section with *Who to follow (*skip *Trending for* this MVP).

- [x]  The *Tweets* section includes:

    1. Simple form for creating a tweet.
    2. List of all tweets (sorted by most recent) that display tweet text and author details.

- [x] The *Who to follow* section includes:

    1. List of profiles that are not followed by the logged-in user (ordered by most recently added).

- [x]  When the user opens the profile page (see the *Userpage* screenshot above), they can see:

    1. Left-side menu (includes only links to pages that are implemented).
    2. Cover picture and *Tweets* tab in the centre (skip other tabs and *Tweet to user* form).
    3. Right-side section with *Profile detailed info.*

- [x]  The *Profile detailed info* section includes:

    1. User photo.
    2. Button to follow a user.
    3. Stats: total number of tweets, number of followers and number of following users.
    4. List of people who follow this user.

- [x] At the end extend your MVP app with one simple feature of your choice.

You should follow the guidelines of the given design, including:

- colors (they are not explicit, you should pick them from design)
- typographies: font face, size and weight (use similar ones to the design)
- layout: composition and space between elements

### CODE REVIEW

**Comment by luqmanmusah** [comment]
Added a Readme file. I will continue to add the other required features.

**Comment by luqmanmusah** [comment]
Made all reviews as suggested. Thank you for the review

File: app/views/devise/mailer/email_changed.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
+
+<% if @resource.try(:unconfirmed_email?) %>
+  <p>We're contacting you to notify you that your email is being changed to <%= @resource.unconfirmed_email %>.</p>
+<% else %>
+  <p>We're contacting you to notify you that your email has been changed to <%= @resource.email %>.</p>
+<% end %>
```
Comment (rloterh) [inline app/views/devise/mailer/email_changed.html.erb:7]: - [ ] There must be no logic found in the views. Move all logic found in views into helper methods.

**Review by rloterh — CHANGES_REQUESTED** [review]
# Changes Required! ♻️
Hi @uqmanmusah!👋 👋
Great job for attempting this capstone project. 👏  However, your project requires lots of refining to make it fit, especially for showcasing on your portfolio. Take time to work on all the changes mandated by the previous reviewers. 

You may schedule a time with a full-time TSE to assist you with all difficulties you are facing with this project.

**Required**
- [ ] Please ensure to implement all changes mandated by previous TSEs.


Just keep working harder 👍 . Kindly make the necessary change(s) and submit another code review request.
Feel free to drop a comment in case of any inquiries about this review. 😉

__Robert Loterh__

**Comment by luqmanmusah** [comment]
These are the review I had from the TSE and I have fixed all of them except removing logic from the view;

- [x]  There are some typos/grammar issues with readme file, kindly use grammarly to spot them out and fix them

- [x]  As the requirements say, you should use PostgreSQL as your database => Using it in your development environment will help you master about configuring posgres db in locat setup

- [x]  There are logics in views, kindly move all of the logic from your views to helper methods

- [x]  The N+1 query problem is no neither talked about in video presentation nor addressed in your codes, kindly fix that issue by explaining it in your video and address it in your codes

- [x]  The project is incomplete, kindly complete it and make sure that your follow the given design with all of the colors

- [x]  There is no test yet, kindly make sure that you test your app by adding both unit and integration tests to your project

- [x]  When a user clicks on cancel my account link on edit user profile page, the app crashes, would you address that issue as well?

- [x]  The live demo link is missing from the readme file, would you deploy your app and add that link?

- [x]  - Ensure to make your ReadMe more descriptive. Your ReadMe is often the first place that the visitors of your repository visit. It must therefore be clean, well structured, detailed, and very easy to comprehend. 

- [x]  I have met all requirements mentioned above.

**Comment by luqmanmusah** [comment]
I have completed all reviews suggested by the TSE's. It is not practical to move the logic from the views page. I have all the hard requirements met for the project. 

I believe I have met all the hard requirements. I would like to get the project approved while I work on the nice-to-have features. 
The views don't work when I move them to the helper.

Thank you.

**Comment by luqmanmusah** [comment]
I have done all review as recommend by the TSE;

- [x] README still has some spelling and grammar errors. KIndly run it through Grammarly to make it error-free.
 I have fixed all the grammatical errors.

- [x] Please add integration tests .

- [x] Left-side menu has a full height in the original design.

 The Profile detailed info section includes:

- [x] User photo.

- [x] Button to follow a user.

- [x] Extend your app with one simple extra feature of your choice and talk about it in the video.
I did this already and I have stated it in my video.

- [x] Please follow the exact design provided.
In the requirement, it said we should follow the layout, colors, and typographies: font face, size, and weight (use similar ones to the design). I believe I have followed the layouts and my design looks similar to the design even though there is a slight difference. 

I believe I have met all the hard requirements of the capstone project. 
I would like to move forward with the curriculum and on my free time and can improve it since I will be adding this project to my portfolio.
Thank you very much.

**Comment by luqmanmusah** [comment]
Fixed the 3 issues mentioned by the previous TSE;

- [x] 1. User should not be able to follow another user more than one time

- [x] Please make use of the assets pipeline to process the style of your page, divide your style into multiple files logically

- [x] When I clicked on the 'people' link from the sidebar and after click the logged-in user's name, it allowed him to follow himself

Thank you for the review

**Comment by luqmanmusah** [comment]
I did the changes but I forgot to push it to heroku. 

I just did that now.

Thank you for the review.

**Pull request merged**